### PR TITLE
New message fetching

### PR DESCRIPTION
**Pull request by rottabonus**
New message fetching implementation

### CODE REVIEW

File: ios/ylitse.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -677,7 +678,7 @@
 				COPY_PHASE_STRIP = NO;
 				ENABLE_STRICT_OBJC_MSGSEND = YES;
 				ENABLE_TESTABILITY = YES;
-				"EXCLUDED_ARCHS[sdk=iphonesimulator*]" = arm64;
+				"EXCLUDED_ARCHS[sdk=iphonesimulator*]" = "";
```
Comment (pheis) [inline ios/ylitse.xcodeproj/project.pbxproj:681]: weird stuff.

File: src/Screens/Main/BuddyList/Button.tsx [hunk 2]
```diff
@@ -20,13 +23,27 @@ const Button = ({ style, buddyId, name, onPress, ...viewProps }: Props) => {
...
+            {name}
+          </RN.Text>
+          <RN.Text
+            style={styles.messageText}
+            ellipsizeMode="tail"
+            numberOfLines={1}
+          >
+            {lastMessage}
+          </RN.Text>
+        </RN.View>
```
Comment (pheis) [inline src/Screens/Main/BuddyList/Button.tsx:46]: oho, last message here too. wow.

File: src/Screens/Main/Chat/MessageList/MemoizedRenderItem.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+  return <Spinner style={styles.spinner} />;
+};
+
+const equalProps = (
+  prevProps: React.ComponentProps<typeof RenderItem>,
+  nextProps: React.ComponentProps<typeof RenderItem>,
+) => prevProps.item.id === nextProps.item.id;
+
+export const MemoizedRenderItem = React.memo(RenderItem, equalProps);
```
Comment (pheis) [inline src/Screens/Main/Chat/MessageList/MemoizedRenderItem.tsx:32]: So fast.

File: src/api/buddies.ts [hunk 4]
```diff
@@ -136,3 +143,21 @@ export function banBuddies(
...
+      resultArray[chunkIndex] = []; // start a new chunk
+    }
+
+    resultArray[chunkIndex].push(item);
+
+    return resultArray;
+  }, []);
+
+  return chunks.map(ch => ({ type: 'InitialMessages', buddyIds: ch }));
+};
```
Comment (pheis) [inline src/api/buddies.ts:163]: What is this?

Comment (pheis) [reply src/api/buddies.ts:163]: too complex for me. Can u explain and let's see how to make it easier to understand.

Comment (pheis) [reply src/api/buddies.ts:163]: ```[...Array(Math.floor(10)).keys()].map(index => buddyIds.slice(index, (index + 1) * 40)).map(chunk => { some: 'lolwut', chunk } );```

Comment (pheis) [reply src/api/buddies.ts:163]: only maps : P

File: src/api/messages.ts [hunk 5]
```diff
@@ -144,3 +198,70 @@ export const readMessage = (buddyId: string) =>
...
+};
+
+export const filterMessages = (
+  buddies: Record<string, Buddy>,
+  messages: MessageMapping,
+) => {
+  const buddyIds = Object.keys(buddies).map(buddyId => buddyId);
+
+  return buddyIds.reduce<MessageMapping>((acc, buddyId) => {
+    const message = messages[buddyId] ? messages[buddyId] : {};
```
Comment (pheis) [inline src/api/messages.ts:228]: ```messages[buddyId] ?? {}```

Comment (pheis) [reply src/api/messages.ts:228]: u can one line this:

```({...acc, [buddyId]: messages[buddyId] ?? {}})```

File: src/api/buddies.ts (hunk 4 above)
Comment (rottabonus) [reply src/api/buddies.ts:163]: image
It looks better to you like this?

Comment (pheis) [reply src/api/buddies.ts:163]: yeah, but (index * 40, (index + 1) * 40)

**Review by idolgov — DISMISSED** [review]

**Review by pheis — DISMISSED** [review]

**Review by idolgov — DISMISSED** [review]

**Review by idolgov — APPROVED** [review]

**Review by pheis — APPROVED** [review]

**Pull request merged**